apprclient.go

```diff
@@ -1 +1,167 @@
+// Package appr holds the client code required to interact with a CNR backend.
 package apprclient
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/url"
+	"path"
+	"time"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/spf13/afero"
+)
+
+// Config represents the configuration used to create a appr client.
+type Config struct {
+	Fs     afero.Fs
+	Logger micrologger.Logger
+
+	Address      string
+	Organization string
+}
+
+// Client knows how to talk with a CNR server.
+type Client struct {
+	fs         afero.Fs
+	httpClient *http.Client
+	logger     micrologger.Logger
+
+	base         *url.URL
+	organization string
+}
+
+// New creates a new configured appr client.
+func New(config Config) (*Client, error) {
+	if config.Fs == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Fs must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.logger must not be empty", config)
```

Comment:
Logger

Reply:
Good catch. Was the same for the settings. Fixed.